src/scene/gsplat/gsplat-instance.js

```diff
@@ -174,6 +174,16 @@ class GSplatInstance {
         const device = this.splat.device;
         viewport[0] = device.width;
         viewport[1] = device.height;
+
+        // adjust viewport for stereoscopic VR sessions
+        if (this.cameras.length > 0) {
+            const camera = this.cameras[0];
```

Comment:
we should probably loop over all cameras to see if any of these is xr. I suspect that if the shadows are used, the cameras[0] would be a shadow camera.

Even better, this should be called per camera somehow, instead of globally once per frame - in case the splat is rendered to a texture as well as main framebuffer.

Reply:
Yeah, it's not great and almost certainly needs to be reworked.